sdk/test/resource/resource_test.cc

```diff
@@ -0,0 +1,138 @@
+#include "opentelemetry/sdk/resource/resource.h"
+#include "opentelemetry/common/key_value_iterable_view.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/common/attribute_utils.h"
+#include "opentelemetry/sdk/resource/resource_detector.h"
+
+#include <cstdlib>
+#include <string>
+#include <unordered_map>
+
+#include <gtest/gtest.h>
+
+class TestResource : public opentelemetry::sdk::resource::Resource
+{
+public:
+  TestResource(opentelemetry::sdk::resource::ResourceAttributes attributes =
+                   opentelemetry::sdk::resource::ResourceAttributes())
+      : Resource(attributes)
+  {}
+};
+
+/*
+class ResourceTest : public testing::Test{
+protected:
+  TestResource resource;
+
+};
+*/
+
+TEST(ResourceTest, create)
+{
+
+  std::map<std::string, std::string> expected_attributes = {
```

Comment:
Can you supplement this with a test that handles strongly-typed attributes of numeric (integer) type?

Reply:
Yes, sure that would be good test to showcase. I will add it.

Reply:
Have modified the test to include both integer and double resource attributes. Please see if this is fine.